src/structure/pdb.jl

```diff
@@ -30,26 +46,314 @@ end
+    downloadpdb(vcat(addedlist,modifiedlist), pdb_dir=pdb_dir, overwrite=true, file_format=file_format)
+    # set the obsolete directory to be inside pdb_dir
+    obsolete_dir=joinpath(pdb_dir,"obsolete")
+    for pdbid in obsoletelist
+        oldfile = joinpath(pdb_dir,"$pdbid"*pdbextension[file_format])
+        newfile = joinpath(obsolete_dir, "$pdbid"*pdbextension[file_format])
+        # if obsolete pdb is in the "pdb_dir", move it to "obsolete" directory inside "pdb_dir" 
+        if isfile(oldfile)
+            if !isdir(obsolete_dir)
+                mkpath(obsolete_dir)
+            end
+            mv(oldfile,newfile)
+        # if obsolete pdb is already in the obsolete directory, inform the user and skip
+        elseif isfile(newfile)
+            info("PDB $pdbid is already moved to the obsolete directory")
+        # if obsolete pdb not available in both pdb_dir and obsolete, inform the user and skip
+        else
+            info("Obsolete PDB $pdbid is missing")
+        end
+    end
+end
+
+
+"""
+Download all obsolete PDB files from RCSB PDB server not present in the local 
+obsolete PDB directory;
+if the keyword argument `obsolete_dir` is set the obsolete PDB files are 
+downloaded to the specified directory;
+"""
+function downloadallobsoletepdb(;obsolete_dir::AbstractString=pwd(), file_format::Type=PDB)
+    # Get all obsolete PDB files in RCSB PDB Server using getallobsolete() and download them
+    obsoletelist = pdbobsoletelist()
+    downloadpdb(obsoletelist, pdb_dir=obsolete_dir, file_format=file_format)
+end
+
+
+"""
+Download and read the PDB file or biological assembly from the RCSB PDB server.
+By default downloads the PDB file to current working directory; 
+if the keyword argument `ba_number` is set the biological assembly with that 
+number will be downloaded; 
+if the keyword argument `pdb_dir` is set the PDB file is downloaded to the 
+specified directory; 
+if the keyword argument `obsolete` is set `true`, the PDB file is downloaded 
+into the obsolete directory inside `pdb_dir`;
+if the keyword argument `overwrite` is set `true`, then it will overwrite the
+PDB file if it exists in the `pdb_dir`;
+"""
+function retrievepdb(pdbid::AbstractString;
+            pdb_dir::AbstractString=pwd(),
+            obsolete::Bool=false, 
+            overwrite::Bool=false, 
+            ba_number::Integer=0,
+            structure_name::AbstractString="$pdbid.pdb",
+            kwargs...)
+    downloadpdb(pdbid, pdb_dir=pdb_dir, obsolete=obsolete, overwrite=overwrite, ba_number=ba_number)
+    info("Parsing the PDB file...")
+    if obsolete
+        # if obsolete is set true, the PDB file will downloaded to obsolete directory inside "pdb_dir"
+        filepath = joinpath(pdb_dir,"obsolete","$pdbid.pdb")
+    else    
+        filepath = joinpath(pdb_dir,"$pdbid.pdb")
+    end
+    pdbpath = ba_number == 0 ? filepath : filepath*"$ba_number"
+    open(pdbpath, "r") do input
```

Comment:
This `open` is not actually required as calling `read` with the filepath is already defined below.